lib/stub_ui/lib/src/ui/text.dart

```diff
@@ -1011,6 +1011,185 @@ enum BoxWidthStyle {
   max,
 }
 
+/// Data class that holds the metrics for a single line in the [Paragraph].
+class LineMetrics {
+
+  @pragma('vm:entry-point')
+  LineMetrics._(
+    this.startIndex,
+    this.endIndex,
+    this.endExcludingWhitespace,
+    this.endIncludingNewline,
+    this.hardBreak,
+    this.ascent,
+    this.descent,
+    this.unscaledAscent,
+    this.height,
+    this.width,
+    this.left,
+    this.baseline,
+    this.lineNumber,
+    List<RunMetrics> rMetrics,
+    List<int> rMetricsIndexes,
+  ) : runMetrics = Map() {
+    for (int i = 0; i < rMetrics.length; ++i) {
+      runMetrics[rMetricsIndexes[i]] = rMetrics[i];
+    }
+  }
+
+  /// The index in the text buffer the line begins.
```

Comment:
(maybe this is clear from context where this will be used, but...)

While reading through these API docs I kept wondering what "the test buffer" is that's mentioned here over and over.